astropy/visualization/wcsaxes/__init__.py

```diff
@@ -3,12 +3,11 @@
 # The following few lines skip this module when running tests if matplotlib is
 # not available (and will have no impact otherwise)
 
-try:
+from ... import tests
+
+if tests.RUNNING:
     import pytest
     pytest.importorskip("matplotlib")
```

Comment:
Wait a minute. Why is this even here? It should have been in the individual test modules under `tests`. Looks like we inherited this from the package we absorbed in #5496. @astrofrog , do you remember why this is at subpackage init level and not the tests?

Reply:
I suspect that this is because wcsaxes is unique in Astropy in that it subclasses many different Matplotlib subclasses, and so the conditional imports of the kind in `astropy.visualization.mpl_normalize` were not appropriate:

```python
try:
    import matplotlib  # pylint: disable=W0611
    from matplotlib.colors import Normalize
    from matplotlib import pyplot as plt
except ImportError:
    class Normalize:
        def __init__(self, *args, **kwargs):
            raise ImportError('matplotlib is required in order to use this '
                              'class.')
```

Reply:
I suppose it was a convenient way to make sure mpl is needed for the whole module and skip it at collection time at the earliest and quickest way possible. Fully agreed that it should not be used like this.